src/javascript/extendable_audio.js

```diff
@@ -0,0 +1,46 @@
+/// <reference path="./extendable_audio.d.ts" />
+
+const SKIP_PROPERTIES = ['dispatchEvent', 'addEventListener', 'removeEventListener'];
+
+export class ExtendableAudio extends DocumentFragment {
+  constructor(url) {
+    this._audio = new Audio();
+    this._documentFragment = document.createDocumentFragment();
```

Comment:
This is basically being used as an EventEmitter, it's what you are actually attaching to when you addEventListener and removeEventListener